src/Reflection/ClassReflection.php

```diff
@@ -709,7 +712,11 @@ public function getActiveTemplateTypeMap(): TemplateTypeMap
 
 	public function isGeneric(): bool
 	{
-		return count($this->getTemplateTags()) > 0;
+		if ($this->isGeneric === null) {
+			$this->isGeneric = count($this->getTemplateTags()) > 0;
```

Comment:
@bastnic can you try profile with change with  `$this->getTemplateTags() !== []` ?

Reply:
considering this code is initializing cache, it likely won't make any visible difference.